include/openthread/border_agent_tracker.h

```diff
@@ -0,0 +1,162 @@
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @addtogroup api-border-agent-tracker
+ *
+ * @brief
+ *   This module includes APIs for the Border Agent Tracker.
+ *
+ * The Border Agent Tracker discovers and tracks Border Agents on the infrastructure link by browsing for the
+ * `_meshcop._udp` mDNS service.
+ *
+ * @{
+ *
+ */
+
+/**
+ * Represents an iterator to iterate through the discovered Border Agents.
+ *
+ * The fields in this struct are for OpenThread internal use only and MUST NOT be accessed or modified by the caller.
+ *
+ * An iterator MUST be initialized using `otBorderAgentTrackerInitIterator()` before it is used.
+ */
+typedef struct otBorderAgentTrackerIterator
+{
+    const void *mPtr;
+    uint64_t    mData;
+} otBorderAgentTrackerIterator;
+
+/**
+ * Represents information about a discovered Border Agent.
+ *
+ * To ensure consistent `mMsecSinceDiscovered` and `mMsecSinceLastChange` time calculations, the iterator's
+ * initialization time is stored within the iterator when `otBorderAgentTrackerInitIterator()` is called. The time
+ * values in this struct are calculated relative to the iterator's initialization time.
+ */
+typedef struct otBorderAgentTrackerAgentInfo
+{
+    const char         *mServiceName;         ///< The service name.
+    const char         *mHostName;            ///< The host name. May be NULL if not known yet.
+    uint16_t            mPort;                ///< The port number. Can be zero if not known yet.
+    const uint8_t      *mTxtData;             ///< The TXT data. May be NULL if not known yet.
+    uint16_t            mTxtDataLength;       ///< The TXT data length.
+    const otIp6Address *mAddresses;           ///< Array of IPv6 addresses of the host. May be NULL if not known yet.
+    uint16_t            mNumAddresses;        ///< Number of addresses in the `mAddresses` array.
+    uint64_t            mMsecSinceDiscovered; ///< Milliseconds since the service was discovered.
+    uint64_t            mMsecSinceLastChange; ///< Milliseconds since the last change  (port, TXT, or addresses).
+} otBorderAgentTrackerAgentInfo;
+
+/**
+ * Enables or disables the Border Agent Tracker.
+ *
+ * Requires `OPENTHREAD_CONFIG_BORDER_AGENT_TRACKER_ENABLE`.
+ *
+ * When enabled, the tracker browses for the "_meshcop._udp" mDNS service to discover and track Border Agents on
+ * the infra-if network.
+ *
+ * @param[in] aInstance A pointer to an OpenThread instance.
+ * @param[in] aEnable   TRUE to enable the Border Agent Tracker, FALSE to disable it.
+ */
+void otBorderAgentTrackerSetEnabled(otInstance *aInstance, bool aEnable);
+
+/**
+ * Indicates whether the Border Agent Tracker is running.
+ *
+ * Requires `OPENTHREAD_CONFIG_BORDER_AGENT_TRACKER_ENABLE`.
+ *
+ * The tracker can be enabled by the user (via `otBorderAgentTrackerSetEnabled()`) or by the OpenThread stack
+ * itself. The tracker is considered running if it is enabled by either entity AND the underlying DNS-SD (mDNS)
```

Comment:
Can we drop "by either entity" since it sounds like those are the only ways to enable?

Reply:
Thanks, Jeff. I think it may help to keep the phrase "by either entity" to be explicit and clear.

My reasoning is that it explicitly states that this is an **OR** condition (i.e., the tracker is running if *either* the user *or* the stack has enabled it). Without this phrase, the term "enabled" might be ambiguous, and someone reading the API docs may assume that it needs to be enabled by both.